Update runtime + game version

user1-github opened the pull request.

AsciiWolf committed:
Update the runtime

AsciiWolf committed:
Add SDL and SDL_image modules

user1-github committed:
Update game + runtime

user1-github committed:
Change to id

user1-github committed:
Update game version in metadata

AsciiWolf commented on net.pistegamez.PekkaKana2.metainfo.xml:
```diff
@@ -25,7 +25,7 @@
   <url type="homepage">https://pistegamez.net/game_pk2.html</url>
   <releases>
-    <release version="1.2.7" date="2020-09-20"/>
+    <release version="1.4.5" date="2023-04-22"/>
```
Wow! I did not realize the game is still being developed. :-) Nice!

It also looks like that they have now switched to a new Git repo and released version 1.5.0:

https://github.com/SaturninTheAlien/pk2_greta/releases/tag/1.5.0

(This repo url is now linked on the [game homepage](https://pistegamez.net/game_pk2.html) as the official one.)

user1-github replied:
Huh, interesting how did I not find this repo in search results..

AsciiWolf replied:
It seems to be a "hard fork", using a slightly different game engine etc. We should probably stay on the older (1.4.5) version at the moment.

user1-github replied:
Ok, I'll leave it on 1.4.5 but it seems the previous repo (from where 1.4.5 is) is also not by the original dev of this game (I don't see versions 1.2.7 or older in it).

AsciiWolf replied:
Ah, you are right! I did not realize that. Then we should probably try updating it to 1.5.0.

AsciiWolf replied:
> Ah, you are right! I did not realize that. Then we should probably try updating it to 1.5.0.

We just need to make sure that the old game config file and save game positions will still be compatible if possible.

user1-github replied:
> > Ah, you are right! I did not realize that. Then we should probably try updating it to 1.5.0.
> 
> We just need to make sure that the old game config file and save game positions will still be compatible if possible.

I have a feeling the main game module in the manifest must be reworked since these new forks look completely different to the old 1.2.7 version. Unfortunately I'm not very familiar with this game, but I'll see what I can do.

SaturninTheAlien replied:
Hi! 🙂

I am the new maintainer of Pekka Kana 2. I came across this discussion recently and would like to clarify a few things.

First of all, thank you for putting Pekka Kana 2 on Flathub.

Danilo’s repository has been abandoned. I originally started my version as a fork because the status of his repository had been unclear for several years. At this point, it is evident that it is no longer maintained, and the project was left in an unstable and buggy state.

PK2 1.4.5 contains several serious issues (including random segmentation faults and incompatibilities with third-party episodes), so it is not recommended for regular play. PK2 1.4.3 is generally considered the last stable release from Danilo’s repository.

My repository later became the official continuation after I received approval from Janne, the original author of PK2. It is based on PK2 1.4.5, but I have already fixed many major issues (the game no longer crashes randomly) and added new features, particularly for mapmakers.

Since most of our communication happens on Discord, the current status of Pekka Kana 2 may be confusing for external maintainers. We apologize for any confusion.

Of course, if you would like me to help with the Flatpak, I would be happy to do so. There was a new stable release (1.5.1) today which adds some improvements related to Linux FHS, so this may help with the Flatpak.

AsciiWolf replied:
Hi Saturnin, thanks for your comment!

It's great to hear that the PK2 development continues. :-) If you want to work on updating this Flatpak, feel free to let me know so I can help with your pull request.

Also, don't hesitate to ask if you want a co-maintainer rights to this Flatpak repository.

SaturninTheAlien replied:
Hi AsciiWolf, thanks!

Co-maintaining this repository would be great.

I've just updated the Flatpak to version 1.5.1. It works fine for me, but it still needs some testing (to make sure it works for others as well 😉).

I'm going to open a new pull request in a minute.

SaturninTheAlien replied:
> We just need to make sure that the old game config file and save game positions will still be compatible if possible.

When talking about the save files, I think I found the source of incompatibility.
In `1.2.7`, the vanilla episode names were stored as were `rooster-island-1` and `rooster-island-2` as the `-` character wasn't rendered by the game. In current versions, these were renamed to `rooster island 1` and `rooster island 2`.

That mismatch appears to break compatibility with old saves, so I prepared a patch for it. I hope to include it in `1.5.1-1`:
https://github.com/SaturninTheAlien/pk2_greta/commit/6d4480dc1322c50ed4bfa26ed00a3d78c7eb7e2b

user1-github committed:
Add include paths

AsciiWolf commented on net.pistegamez.PekkaKana2.yaml:
```diff
@@ -39,6 +39,10 @@ modules:
     buildsystem: simple
+    build-options:
+      env:
+        C_INCLUDE_PATH: /app/include/SDL2:/app/include
```
I wonder why this is needed to be specified manually when using the newer runtime.

user1-github replied:
I explained this [here](https://github.com/flathub/org.neverball.Neverball/pull/9), see my last 2 or 3 comments. It's the same workaround I had to do for Neverball yesterday, for Sauerbraten and a few other games as well when updating them to 25.08.

AsciiWolf replied:
Ah, thanks for the info!

user1-github committed:
Update project license and game version to 1.5.0

The license is from the newest game fork Github repo

user1-github committed:
Update .desktop file from Debian

it looks better

user1-github committed:
Try updating to 1.5.0

user1-github committed:
Add no-autogen: true

user1-github committed:
Fix typo

user1-github committed:
Add lua shared module and try a different way

user1-github committed:
Add libzip

user1-github committed:
Add buildsystem: cmake-ninja to libzip

user1-github committed:
Add cleanup section and include paths

user1-github committed:
Update include paths

user1-github committed:
Try building with lua 5.3

user1-github committed:
Last try: lua 5.1

user1-github committed:
Add lua 5.4 back

user1-github commented:
> In file included from src/lua/pk2_lua.hpp:13,
                 from src/episode/episodeclass.cpp:23:
   src/3rd_party/sol.hpp:3110:177: error: operator ‘!=’ has no left operand
   3110 |         #if (SOL_LUA_VERSION_I_ == 502) || (defined(LUA_COMPAT_BITLIB) && (LUA_COMPAT_BITLIB != 0)) ||  (SOL_LUA_VERSION_I_ < 504 && (defined(LUA_COMPAT_5_2) && (LUA_COMPAT_5_2 != 0)))
      |                                                                                                                                                                                 ^~
src/3rd_party/sol.hpp: In member function ‘T& sol::optional<T&>::emplace(Args&& ...)’:
src/3rd_party/sol.hpp:6755:31: error: ‘class sol::optional<T&>’ has no member named ‘construct’ [-Wtemplate-body]
 6755 |                         this->construct(std::forward<Args>(args)...);
      |                               ^~~~~~~~~
make: *** [Makefile:89: build/episode/episodeclass.o] Error 1
Error: module pekkakana2: Child process exited with code 2
error: Recipe `build` failed with exit code 1
Error: Process completed with exit code 1.

user1-github commented:
oops, accidentally closed this but I have no idea how to fix the error above. It seems like some incompatibility between a third party library that version 1.5.0 uses and any lua version available on Flathub.

user1-github committed:
Add -std=c++17 cxxflags

not sure if it will fix the build error, but it's worth trying

user1-github committed:
Try version 1.4.5 again

user1-github committed:
Revert to 1.4.5

user1-github committed:
Fix typo in post-install command

user1-github committed:
Change exec

user1-github committed:
Create symbolic link in /app/bin that points to the game binary

user1-github commented:
@AsciiWolf It seems impossible to use version 1.5.0 due to the error a few comments above (looks like an upstream issue), but now I switched to version 1.4.5 and it seems to build fine.
But I'm not sure about how the binary is supposed to be launched (I just created a symbolic link in /app/bin that points to the game binary), where the game binary is supposed to be relatively to the game assets and whether `--persist=.pekka-kana-2` is still needed.
I can't test that myself atm, so you'll have to check it.

user1-github committed:
Add --persist=.pekka-kana-2 and potential dir correction

user1-github committed:
Remove lua and libzip deps

They don't seem to be needed for version 1.4.5

user1-github committed:
Switch back to simple buildsystem

user1-github committed:
Create /app/bin before creating symlink there

I have no idea why does it suddenly complain about no such file or directory

user1-github committed:
Cleanup some redundancies

AsciiWolf commented on net.pistegamez.PekkaKana2.desktop:
```diff
@@ -1,9 +1,9 @@
-Comment=Jump 'n run game made in the spirit of old classic platformers
-Icon=net.pistegamez.PekkaKana2
-Exec=pekka-kana-2.sh
+Comment=2D Oldschool platform game where you control a rooster named Pekka
```
To be honest, I prefer the original description. :) And it also matches the AppStream metainfo summary.

user1-github replied:
Ok, will change. To me they both look fine, I just borrowed the .desktop changes from the Debian package, cause I thought it looks better in general.

AsciiWolf replied:
No worries. They both look good. :)

AsciiWolf commented on net.pistegamez.PekkaKana2.yaml:
```diff
@@ -1,63 +1,40 @@
+  - shared-modules/SDL2/SDL2_image.json
 
-  - name: pekkakana2
+  - name: pekka-kana-2
```
Why changing this? Flatpak modules usually have a naming scheme that I attempt to follow and one of the unwritten rules is that the name is as simple as possible, without any special characters. The `pekkakana2` one is good enough.

Also, I am not a fan of adding unrelated changes to the PR.

user1-github replied:
Ok, sorry, will revert and won't do unrelated changes. I just thought it looks more clear this way. Though if a module name consists of more than one word, I personally don't see what's the issue with separating them with dashes. I mean it's a common practice on Flathub. For example see [org.winehq.Wine](https://github.com/flathub/org.winehq.Wine/blob/branch/stable-25.08/org.winehq.Wine.yml#L387) and [bottles](https://github.com/flathub/com.usebottles.bottles/blob/master/com.usebottles.bottles.yml#L117) manifests.

AsciiWolf commented on net.pistegamez.PekkaKana2.yaml:
```diff
@@ -1,63 +1,40 @@
+      - install -Dm755 ./bin/pekka-kana-2 /app/share/pekka-kana-2/pekka-kana-2
+      - cp -a res /app/share/pekka-kana-2
+      - mkdir -p /app/bin
+      - ln -s /app/share/pekka-kana-2/pekka-kana-2 /app/bin/pekka-kana-2
```
To be honest, I don't like this whole symlink solution and also the `pekka-kana-2` directory created directly in `/app/share`. The `/app/share/games/` prefix was the correct one as specified by the XDG directory spec.

user1-github replied:
Unlike version 1.2.7, version 1.4.5 doesn't have any .sh script that you install into /app/bin. So honestly, I don't know what else can be done here. If I'll directly install the binary into /app/bin, it probably won't find the game assets (in the res folder) and if I don't install anything into /app/bin, the build will fail.

Regarding the dir created directly in /app/share, I didn't think it's an issue due to other games on Flathub (like Xonotic, Red Eclipse, Sauerbraten, etc) also having their dirs directly in /app/share.
But ok, will change it back.

user1-github replied:
Btw, did you test if the last build works? Cause the main thing I'm currently not sure about is where exactly does the binary need to be relatively to the game assets (or the res dir). I'm away from my computer atm, so can't try it myself.

user1-github replied:
I just found Debian (which is also on Version 1.4.5) uses [this patch](https://salsa.debian.org/games-team/pekka-kana-2/-/blob/master/debian/patches/03-datapath.patch?ref_type=heads) that hardcodes the game data path to /usr/share/games/pekka-kana-2/res. I modified it to /app/share/games/pekka-kana-2/res, so this way the binary can be directly installed into /app/bin and now we don't need the symlink anymore.
The last test build includes this change + the other changes you suggested.

AsciiWolf replied:
> Btw, did you test if the last build works?

Sorry, not yet. :( I am currently on a vacation and my time is limited. I will take a look later.

Thanks for your great work on this!

user1-github replied:
My pleasure :)
Make sure to test the latest build with all the latest changes.
Enjoy your vacation :)

AsciiWolf replied:
Will do, thanks! :)

AsciiWolf commented on net.pistegamez.PekkaKana2.yaml:
```diff
@@ -1,63 +1,40 @@
+      - make
+      - install -Dm755 ./bin/pekka-kana-2 /app/share/pekka-kana-2/pekka-kana-2
+      - cp -a res /app/share/pekka-kana-2
+      - mkdir -p /app/bin
```
This should not be needed. `/app/bin` is a standard directory, just like `/app/share` etc. On the other hand, it does no harm to have it there, so you can probably keep it.

user1-github replied:
I know, but something's weird is going on with Flatpak builder. At some point the build has started failing due to no such file or directory error for some reason. So I had to explicitly create /app/bin.

Edit: it seems this is not needed anymore when installing the binary directly into /app/bin.

AsciiWolf commented on net.pistegamez.PekkaKana2.yaml:
```diff
@@ -1,63 +1,40 @@
-        for dir in episodes gfx language music sfx sprites; do
-          cp -r data/$dir /app/share/games/pekka-kana-2/$dir
-        done
+      - make
```
```suggestion
      - make -j $FLATPAK_BUILDER_N_JOBS
```

user1-github replied:
Will do.

AsciiWolf commented on net.pistegamez.PekkaKana2.desktop:
```diff
@@ -1,9 +1,9 @@
 Type=Application
+Icon=net.pistegamez.PekkaKana2
 Categories=Game;AdventureGame;ActionGame;
-PrefersNonDefaultGPU=true
```
I am not a huge fan of removing this. The default GPU handling is an important thing for multi GPU systems even though its current implementation is not ideal. It is already being sorted out in upstream (both KDE and GNOME).

AsciiWolf replied:
On the other hand, it is probably not much useful to have it there for such simple and lightweight game. So, feel free to keep it removed.

user1-github replied:
Yeah, since this is a very simple 2D game, I thought it makes more sense to run it on an iGPU for power efficiency.

user1-github committed:
Upload datapath.patch

user1-github committed:
Add back the previous .desktop comment

user1-github committed:
Add datapath patch + apply suggested changes

user1-github committed:
Add screenshot captions in metadata

AsciiWolf commented on net.pistegamez.PekkaKana2.metainfo.xml:
```diff
@@ -18,9 +18,11 @@
   <screenshots>
     <screenshot type="default">
       <image>https://raw.githubusercontent.com/flathub/net.pistegamez.PekkaKana2/master/screenshots/screenshot1.jpg</image>
+      <caption>Gameplay 1</caption>
```
This is not needed. The caption warnings are just warnings and "Gameplay 1" is not an useful caption. :)

user1-github replied:
I remember in some other PR you asked me to fix the linter warnings (was the same issue with missing captions), that's why I added them here.

AsciiWolf replied:
It was probably another linter warning. There's no point in having such generic captions in the metadata.

user1-github committed:
Remove screenshot captions

AsciiWolf commented:
The test build version sadly does not run:

```
$ flatpak run net.pistegamez.PekkaKana2//test
[DEBUG]	PFile	- Find ./data/redirect.txt
[INFO]	PFile	- ./data/redirect.txt not found, trying different cAsE
[WARN]	PFile	- Couldn't scan on "./data"
[INFO]	PFile	- ./data/redirect.txt not found
[ERR]	PFile	- Can't get RW from file "./data/log.txt"
[DEBUG]	PK2	- Pekka Kana 2 started!
[DEBUG]	PK2	- Game version: Pekka Kana 2 Definitive Edition 1.4.5 (Linux) (no-zip) (portable)
[DEBUG]	PK2	- Number: 0x405
[DEBUG]	PK2	- Cammit hash: 
[DEBUG]	PK2	- Data path - ./data/
[ERR]	PFile	- Can't get RW from file "./data/settings.ini"
[DEBUG]	PK2	- Searching language from code: cs
[ERR]	PFile	- Can't get RW from file "./data/settings.ini"
[ERR]	PK2	- Can't save settings
[ERR]	PFile	- Can't get RW from file "./data/config.txt"
[ERR]	PFile	- Can't get RW from file "./data/config.txt"
[DEBUG]	Piste	- We compiled against SDL version 2.32.56 ...
[DEBUG]	Piste	- But we are linking against SDL version 2.32.56.
[DEBUG]	PDraw	- Initializing buffers
[DEBUG]	PRender	- Initializing graphics
[DEBUG]	PRender	- Video driver: wayland
[DEBUG]	PRender	- Loading OpenGL functions
[DEBUG]	PRender	- OpenGL version: 4.60
[DEBUG]	PRender	- GLSL version: 4.6 (Compatibility Profile) Mesa 25.2.4 (git-ab462ae6b7)
[INFO]	PInput	- No haptic found
[INFO]	PInput	- No Controller found
[DEBUG]	PSound	- buffer size: 1024
[DEBUG]	PSound	- Desired 44100hz 0x8010
[DEBUG]	PSound	- Audio driver: pulseaudio
[DEBUG]	PSound	- Got 44100hz 0x8010
[DEBUG]	PK2	- Loading language from language/czech.txt
[DEBUG]	PFile	- Find language/fonts/ScandicSmall.txt
[DEBUG]	PFile	- Found on language/fonts/ScandicSmall.txt
[DEBUG]	PFile	- Find language/fonts/ScandicSmall.bmp
[DEBUG]	PFile	- Found on language/fonts/ScandicSmall.bmp
[DEBUG]	PDraw	- Created font from language/fonts/ScandicSmall.txt - id 0
[DEBUG]	PFile	- Find language/fonts/ScandicBig1.txt
[DEBUG]	PFile	- Found on language/fonts/ScandicBig1.txt
[DEBUG]	PFile	- Find language/fonts/ScandicBig.bmp
[DEBUG]	PFile	- Found on language/fonts/ScandicBig.bmp
[DEBUG]	PDraw	- Created font from language/fonts/ScandicBig1.txt - id 1
[DEBUG]	PFile	- Find language/fonts/ScandicBig2.txt
[DEBUG]	PFile	- Found on language/fonts/ScandicBig2.txt
[DEBUG]	PFile	- Find language/fonts/ScandicBig.bmp
[DEBUG]	PFile	- Found on language/fonts/ScandicBig.bmp
[DEBUG]	PDraw	- Created font from language/fonts/ScandicBig2.txt - id 2
[DEBUG]	PFile	- Find language/fonts/ScandicBig3.txt
[DEBUG]	PFile	- Found on language/fonts/ScandicBig3.txt
[DEBUG]	PFile	- Find language/fonts/ScandicBig.bmp
[DEBUG]	PFile	- Found on language/fonts/ScandicBig.bmp
[DEBUG]	PDraw	- Created font from language/fonts/ScandicBig3.txt - id 3
[DEBUG]	PFile	- Scanned on "language" for ".txt". Found 37 matches
[DEBUG]	PFile	- Scanned on "episodes" for "/". Found 2 matches
[DEBUG]	PK2	- Found 2 episodes
[DEBUG]	Piste	- FPS set to 60
[WARN]	PSound	- Unrecognized audio format
[ERR]	PFile	- Can't get RW from file "./data/saves.dat"
[INFO]	PK2	- No save file
[ERR]	PFile	- Can't get RW from file "./data/saves.dat"
[ERR]	PK2	- Can't save records
[WARN]	PSound	- Unrecognized audio format
[ERR]	PK2	- Can't load intro.xm
[DEBUG]	PGl	- Terminated
[DEBUG]	PK2	- Terminated
[DEBUG]	PLog	- Terminated
```

It is probably caused by the missing modplug support:

```
[WARN]	PSound	- Unrecognized audio format
[ERR]	PK2	- Can't load intro.xm
```

user1-github commented:
It seems SDL2_mixer currently included in the SDK is [compiled without modplug support](https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/blob/master/elements/components/sdl2-mixer.bst?ref_type=heads#L17) for some reason so I guess we need to add back the bundled SDL2_mixer and modplug deps (I originally thought they were leftovers).

But it seems some errors are unrelated to that. Like this one: `[ERR]	PFile	- Can't get RW from file "./data/log.txt"` Debian uses [3 more patches](https://salsa.debian.org/games-team/pekka-kana-2/-/tree/master/debian/patches?ref_type=heads) apart from the datapath.patch that I've already added. Idk if the 1st and 4th patch are going to be useful, but I suspect maybe the 2nd one will fix the other errors? Or maybe I should just try to add all of these patches?

AsciiWolf commented:
Please add just the SDL2_mixer and modplug for now. We can figure out the rest later. Thanks!

user1-github committed:
Upload dont-portable.patch

user1-github committed:
Add dont-portable.patch and add back bundled deps

The patch should fix user data related errors. The bundled SDL2_mixer and libmodplug deps are still needed because SDL2_mixer that is in the SDK is compiled without modplug support.

user1-github commented:
> Please add just the SDL2_mixer and modplug for now. We can figure out the rest later. Thanks!

I've also added the dont-portable.patch from Debian because it seems like without this patch the game will try to write user data into the res dir under /app/share/games/pekka-kana-2 (which is obviously unwritable).

user1-github committed:
Also add back cleanup section due to the bundled deps

user1-github committed:
Remove unneeded cleanup items

AsciiWolf commented on net.pistegamez.PekkaKana2.metainfo.xml:
```diff
@@ -25,7 +25,7 @@
   </screenshots>
   <url type="homepage">https://pistegamez.net/game_pk2.html</url>
   <releases>
-    <release version="1.2.7" date="2020-09-20"/>
```
We can keep the 1.2.7 entry and just add the 1.4.5 one.

user1-github replied:
Oops, yeah you're right, I'll add back the previous version entry. Don't know why I removed it.

AsciiWolf commented:
The game now runs and seems to work fine, thanks! :-)

I have however noticed two things:

1. The save games and settings from older version are not kept. We should probably add some migration script (similar to what for example Minecraft used to have).
2. I have noticed that the ingame NPCs now act quite strange. There may be some issue with their AI in the new version.

I will do more testing later...

user1-github commented:
> The save games and settings from older version are not kept. We should probably add some migration script (similar to what for example Minecraft used to have).

Then we need to know where did version 1.2.7 store the saves and settings and where does version 1.4.5 stores them now (though I'm not sure how much I'll be able to help here as I'm not very experienced in Bash). Also, if the new version keeps the saves and settings in a different directory, it might also be possible that it uses a different format that is not backwards compatible with 1.2.7. I think you should look at how do the savegame and settings files look exactly in both versions since I haven't found any info about savegame backwards compatibility between different versions of the game.

user1-github committed:
Add previous version to metadata

AsciiWolf commented:
> Then we need to know where did version 1.2.7 store the saves and settings and where does version 1.4.5 stores them now (though I'm not sure how much I'll be able to help here as I'm not very experienced in Bash). Also, if the new version keeps the saves and settings in a different directory, it might also be possible that it uses a different format that is not backwards compatible with 1.2.7. I think you should look at how do the savegame and settings files look exactly in both versions since I haven't found any info about savegame backwards compatibility between different versions of the game.

Old version:

```
[📦 net.pistegamez.PekkaKana2 ~]$ pwd
/home/asciiwolf
[📦 net.pistegamez.PekkaKana2 ~]$ ls -la .pekka-kana-2/
celkem 8
drwxr-xr-x. 1 asciiwolf asciiwolf   74 29. říj 22.15 .
drwx------. 5 asciiwolf asciiwolf  100 29. říj 22.15 ..
drwxr-xr-x. 1 asciiwolf asciiwolf   20 29. říj 22.15 rooster-island-1
-rw-r--r--. 1 asciiwolf asciiwolf 2610 29. říj 22.15 saves.dat
-rw-r--r--. 1 asciiwolf asciiwolf  204 29. říj 22.15 settings.ini
```

New version:

```
[📦 net.pistegamez.PekkaKana2 ~]$ pwd
/home/asciiwolf
[📦 net.pistegamez.PekkaKana2 ~]$ ls -la .var/app/net.pistegamez.PekkaKana2/data/Pekka\ Kana\ 2/
celkem 40
drwx------. 1 asciiwolf asciiwolf   104 29. říj 22.16 .
drwxr-xr-x. 1 asciiwolf asciiwolf    24 29. říj 22.16 ..
-rw-r--r--. 1 asciiwolf asciiwolf   467 29. říj 22.16 config.txt
-rw-r--r--. 1 asciiwolf asciiwolf 28141 29. říj 22.17 log.txt
drwx------. 1 asciiwolf asciiwolf     0 29. říj 22.16 mapstore
-rw-r--r--. 1 asciiwolf asciiwolf  2837 29. říj 22.17 saves.dat
drwx------. 1 asciiwolf asciiwolf    40 29. říj 22.17 scores
-rw-r--r--. 1 asciiwolf asciiwolf   226 29. říj 22.16 settings.ini
```

But you are right that the format is different and does not seem to be compatible. So, there is no point in adding any migration of the old data.

AsciiWolf commented:
One more bug: The keyboard (arrow keys) movement in the main menu is too fast when using the new version. It works fine on the older version.

AsciiWolf commented on net.pistegamez.PekkaKana2.yaml:
```diff
@@ -1,24 +1,24 @@
   - --socket=wayland
   - --socket=fallback-x11
   - --socket=pulseaudio
   - --persist=.pekka-kana-2
```
I can confirm that this dir is no longer used in the new version and the persist rule can be removed.

user1-github committed:
Remove --persist=.pekka-kana-2
